hnn_core/network.py

```diff
@@ -518,3 +518,32 @@ def plot_spikes(self, ax=None, show=True):
         if show:
             plt.show()
         return ax.get_figure()
+
+    def write_spikes(self, fname, trial_idx='all'):
+        """Write spike times to a file.
```

Comment:
how do you plan to read them back in?

Reply:
Check out the plot_simulate_evoked.py script in the attached demo.zip folder. Are you thinking we should have a built-in read function as well?

Reply:
HNN has the functionality to take a given parameter file, find the spike output in an expected directory and load it. If this functionality is part of hnn-core, HNN can use it. Otherwise, the function will still be part of HNN.

Reply:
So are you in favor of adding a read function in hnn-core or would you rather that be handled by an HNN wrapper?

Reply:
I think you should add a reader if you're going to have a way to write it :) It's also easier to test that way. You can do a round trip and check if you have what you started with.

Reply:
Since spikes currently exist only as a `network` class attribute, they either need to be read in through a `network` class method at/after network initialization or handled as their own object outside of the `network` class (as with params). In keeping with our desire to make HNN data structures more transparent and tractable (e.g., using .json files for params, json schema for param validation), what do you think of turning spikes into their own object with the following example structure:
```
{
    <gid_type>: {
        <gid>: [
            <spiketime[0]>,
            <spiketime[1]>,
            ...
            ]
        }
}
```
We could still maintain the .txt file format convention of HNN output for now, but this would make the spike data more intuitive, modular, and readable into a `network` instance with proper tests and validation.

Reply:
Humm ... I would keep it simple for now. You can read in a dictionary of lists. The dictionary would contain three keys `gid`, `gid_type` and `spiketimes`. It sounds like this stuff should belong to the `_Cell` class but I'm sure it's nasty to save it directly and read it back. Let's stick to python built-ins until it becomes apparent if something should go to a separate class.

Reply:
thinking more, you could probably use a list of [NamedTuple](https://docs.python.org/2/library/collections.html#collections.namedtuple). Maybe that simplifies some code, I don't know ... would keep it for another PR.

Reply:
Yeah, a reader would be helpful. In the future, I like the idea of moving spiketime and spikegids out of the network class. I'm in favor of either keeping things as close to HNN for now or the other extreme of making a new Spike or Spikedata class.

Reply:
The namedtuple class does look super convenient for this - good call. In the short term, I'll just implement a simple read function that validates gids prior to them overwriting the spikes of a `network` instance (to be compatible with the ranges in `network.gid_dict`).